apps/studio/src/schemas/user.ts

```diff
@@ -50,3 +50,26 @@ export const updateInputSchema = z.object({
 })
 
 export const updateOutputSchema = z.boolean()
+
+export const updateDetailsInputSchema = z.object({
+  name: z.string().trim().min(1, "Name is required"),
+  phone: z
+    .string()
+    .trim()
+    .min(1, "Phone number is required")
+    .refine((phone) => {
+      const cleanPhone = phone.replace(/\s+/g, "")
+      return !isNaN(Number(cleanPhone)) && cleanPhone.length === 8
+    }, "Phone number must be exactly 8 digits")
+    .refine((phone) => {
+      const cleanPhone = phone.replace(/\s+/g, "")
+      return (
+        cleanPhone.startsWith("6") ||
+        cleanPhone.startsWith("8") ||
+        cleanPhone.startsWith("9")
+      )
+    }, "Phone number must start with 6, 8, or 9")
+    .transform((phone) => phone.replace(/\s+/g, "")), // Clean up any remaining whitespace
+})
```

Comment:
[FYI] i think zod has a `transform` method that might be exactly what you're looking for! [This](https://zod.dev/?id=relationship-to-refinements) allows us to do the removal of whitespace one and makes the pipeline clearer

Reply:
ah thanks for sharing, didn't know about this! fixed in https://github.com/opengovsg/isomer/pull/1122/commits/552f4792e4bf7d36f493e6533899f32b5687751a